packages/react-server/src/ReactFlightReplyServer.js

```diff
@@ -397,6 +409,23 @@ function parseModelString(
           key,
         );
       }
+      case 'K': {
+        // FormData
+        const stringId = value.substring(2);
+        const formPrefix = response._prefix + stringId + '_';
+        const data = new FormData();
+        const backingFormData = response._formData;
+        // We assume that the reference to FormData always comes after each
+        // entry that it references so we can assume they all exist in the
+        // backing store already.
+        // $FlowFixMe[prop-missing] FormData has forEach on it.
+        backingFormData.forEach((entry: File | string, entryKey: string) => {
+          if (entryKey.startsWith(formPrefix)) {
+            data.append(entryKey.substr(formPrefix.length), entry);
```

Comment:
substr is kinda deprecated? annex B.

```suggestion
            data.append(entryKey.substring(formPrefix.length), entry);
```

(personally I'm a slice gal)

Reply:
I'd be down to just lint and enforce a single form just for the potential compression savings by using a repeated string.